h2/src/main/org/h2/engine/Mode.java

```diff
@@ -198,6 +198,11 @@
      */
     public HashMap<String, DataType> typeByNameMap = new HashMap<>();
 
+    /**
+     * Whether not to allow C-style //... comments.
+     */
+    public boolean disallowCStyleLineComments;
```

Comment:
`//` is a C++-style comment, not the C-style. C-style comment is `/* */`. This name is not correct and should be replaced with `CppStyle`.

It also looks like this flag should be reversed to allow C++-style comments instead. Because only H2 (REGULAR mode) and HSQLDB allows such comments. So it's simpler to set this flag to `true` in this two modes.

Reply:
Okay, thanks. Will rename to `allowCppStyleComments` and reverse the meaning.

Reply:
If you are certain that we will not break existing code where // would actually be allowed, then that's fine. So if you are 100% certain that it is really only H2 and HSQLDB, then I will change it.
The reason for using "disallow" with false as default was just to not risk breaking existing code, other than PostgreSQL where I know it does not support // comments.

Reply:
* SQL Server: Incorrect syntax near '/'.
* Oracle: ORA-00900: invalid SQL statement
* MySQL: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '//' at line 1
* DB2 documentation: https://www.ibm.com/support/knowledgecenter/en/SSEPEK_12.0.0/sqlref/src/tpc/db2z_sqlcomments.html
* Derby documentation: https://db.apache.org/derby/docs/10.14/tools/rtoolsijcomref16653.html
* Ignite documentation: https://apacheignite-sql.readme.io/docs/sql-conformance (E161 is supported that means -- comments). I didn't test it, but there are no other information about allowed comments in its documentation.

HSQLDB, however, have such comments:
http://www.hsqldb.org/doc/guide/ch09.html#N124ED